ctrl+h to hide/show hidden files

a-rose opened the pull request:
Hi,

I added a keybind to toggle displaying hidden files and filers (starting with '.').
Hidden files and folders are still displayed by default.

a-rose committed:
Added a keybind (ctrl+h) to toggle displaying hidden files and folders

s-a commented:
Looks good. I'll take a closer look tomorow. I suggest to save this setting permanent in toggle function and initialize from settings object? Some example code from my memory...
```
applicationController.config.settings.showHiddenFiles = boolean
applicationController.config.save();
```

You can simply add more commits to your branch. I think this will update your PR automatically.

2 things this function could realy need. 
- [ ] indicator to display in GUI which mode is currently active. I would wonder if i could not find files and did not remember I toggled mode.
- [ ] take care of file/folder attribute `hidden` which should bei a part of filesystem object already available in code.

a-rose committed:
index.js: simplified toggleShowHiddeFiles()

Simplified toggleShowHiddeFiles() according to this comment: https://github.com/a-rose/sulu-file-system-view/commit/33fcb9c82b0b14d5fe42460feae7318305748b1a#diff-168726dbe96b3ce427e7fedce31bb0bc

a-rose committed:
Update index.js

a-rose commented:
I updated index.js with your suggestion to simplify the toggle.

Regarding your other remarks:

- Storing the state in the config: agreed ! I didn't know where I should do it.
- Displaying the current state on the GUI: agreed too ! I tried to add it but I couldn't find the right way to do it, I need to investigate more.
- "take care of file/folder attribute hidden which should bei a part of filesystem object already available in code": I'll look into this as well.

Thanks for your comments :+1:

s-a commented on index.js:
```diff
@@ -36,6 +36,15 @@ Command.prototype.historyJumpBackward = function historyJumpBackward() {
 };
 
+
+Command.prototype.toggleShowHiddeFiles = function toggleShowHiddeFiles() {
```
Here is a small typo `toggleShowHiddenFiles`

s-a commented on index.js:
```diff
@@ -36,6 +36,15 @@ Command.prototype.historyJumpBackward = function historyJumpBackward() {
+Command.prototype.toggleShowHiddeFiles = function toggleShowHiddeFiles() {
+	var view = this.GUI.activeView().model;
+	view.showHiddenFiles = !view.showHiddenFiles;
+	view.updateGridViewData(false);
```
To save state: Insert this code here: 
```javascript
applicationController.config.settings.showHiddenFiles = view.showHiddenFiles;
applicationController.config.save();
```

If we want to save state per view use:
`applicationController.config.settings.fileSystemViews[this.id]`

It is confusing and hard to get because `applicationController` is global :/. See here for API https://github.com/s-a/user-appdata#methods. I created an issue here https://github.com/sulu-one/sulu/issues/42

s-a commented on file-system-view.js:
```diff
@@ -17,6 +17,7 @@ var View = function(id) {
 	this.path = cfg.path;
 	this.sep = require("path").sep;
 	this.history = [this.path];
+	this.showHiddenFiles = true;
```
To init toggle state use `applicationController.config.settings.showHiddenFiles`

s-a commented on file-system-view.js:
```diff
@@ -243,10 +244,15 @@ View.prototype.renderRow = function(fileSystemItem, showFullPath) {
 	for (var i = 0; i < fileSystemItems.length; i++) {
 		var fileSystemItem = fileSystemItems[i];
+
+		if(showHiddenFiles === false && fileSystemItem.name[0] === '.' && fileSystemItem.name !== ".."){
```
To check filesystemitem` s hidden state check `fileSystemItem.mode` See https://nodejs.org/api/fs.html#fs_class_fs_stats. It is a integer which needs to be interpreted by code. I guess this is a unix code which describes item`s state on file system.

s-a commented:
Just added a few comments. Please keep in mind I suggested code from memory. So it is not 	
inevitably correct 😺 . Regarding the state display in GUI : 
There is an example here https://github.com/sulu-one/sulu-file-system-view/blob/5237d57c76203fc7d5f353a10d31229756feb073/file-system-view.js#L128 and https://github.com/sulu-one/sulu-file-system-view/blob/5237d57c76203fc7d5f353a10d31229756feb073/element-data-view.html#L35 which updates the `selectedFileCount` when you press `space`. SULU makek use of Google` s Polymer https://www.polymer-project.org/1.0/docs/devguide/feature-overview which supports two way binding.

a-rose committed:
Fixed typo: toggleShowHiddeFiles=> toggleShowHiddenFiles

a-rose committed:
Added showing the current value of showHiddenFiles on the UI

a-rose committed:
Fix title in readme

a-rose committed:
Changed showHiddenFilesState to use a FontAwesome icon instead of text, with a tooltip displaying state + nb of hidden files
